sympy/concrete/tests/test_sums_products.py

```diff
@@ -833,7 +833,11 @@ def test_matrix_sum():
     assert Sum(A,(n,0,3)).doit() == Matrix([[0, 4], [6, 0]])
 
 
-def test_issue_9594():
+def test_indexed_idx_sum():
     i = symbols('i', cls=Idx)
     r = Indexed('r', i)
-    Sum(r, (i, 0, 3)).doit() == [r.subs(i, j) for j in range(4)]
+    assert Sum(r, (i, 0, 3)).doit() == sum([Indexed('r', Idx(j)) for j in range(4)])
+
+    k = Idx('k', range=(1, 3))
+    A = IndexedBase('A')
+    assert Sum(A[k], k).doit() == sum([A[Idx(j, (1, 3))] for j in range(1, 4)])
```

Comment:
```
In [6]: Sum(A[k], k).doit() == sum([A[Idx(j, (1, 3))] for j in range(1, 4)])
Out[6]: True

In [7]: Sum(A[k], k).doit() == sum([A[Idx(j)] for j in range(1, 4)])
Out[7]: False

In [8]: Sum(A[k], k).doit() == sum([A[j] for j in range(1, 4)])
Out[8]: False
```

It appears to be a bug. I would suggest that `Idx(number)` should return a _number_ (that is, no _Idx_ object) to solve this issue.